src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -853,8 +853,16 @@ class TableInlineConstraintSegment(BaseSegment):
     """Inline table constraint for CREATE / ALTER TABLE."""
 
     type = "table_constraint_definition"
-    match_grammar = Sequence(
-        Sequence("CONSTRAINT", Ref("SingleIdentifierGrammar"), optional=True),
+    match_grammar = StartsWith(
+        OneOf("CONSTRAINT", "NOT", "NULL", "PRIMARY", "FOREIGN"),
+        terminator=OneOf("COMMENT", Ref("CommaSegment"), Ref("EndBracketSegment")),
+    )
+    parse_grammar = Sequence(
+        Sequence(
+            "CONSTRAINT",
+            Ref("NakedIdentifierSegment", optional=True),
+            optional=True,
+        ),
```

Comment:
The problem is it doesn't know the next keyword (PRIMARY) is not the optional naked identifier (since they are UNRESERVED keywords so could be used as identifiers). 

However I recently learned off a clever way around this, by using the `excludes` function (only available in `AnyNumberOf` so have to use that and then limit it to be 0 or 1, which is same as `Ref(..., optional=True):

```suggestion
    parse_grammar = Sequence(
        Sequence(
            "CONSTRAINT",
            AnyNumberOf(
                Ref("NakedIdentifierSegment"),
                max_times=1,
                min_times=0,
                exclude=OneOf("NOT", "NULL", "PRIMARY", "FOREIGN"),
            ),
            optional=True,
        ),
```